backend/src/main/java/de/tk/mastermind/controller/PlayerController.java

```diff
@@ -1,31 +1,47 @@
 package de.tk.mastermind.controller;
 
+import de.tk.mastermind.models.RegistrationData;
 import de.tk.mastermind.models.Player;
 import de.tk.mastermind.service.PlayerService;
 import lombok.RequiredArgsConstructor;
 
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.web.bind.annotation.*;
 
+import java.security.Principal;
+import java.util.Optional;
+
 @RestController
 @CrossOrigin
 @RequiredArgsConstructor
 @RequestMapping("/api/players")
 public class PlayerController {
 
     private final PlayerService playerService;
+    private final PasswordEncoder passwordEncoder;
 
     @PostMapping
-    public Player createPlayer(@RequestBody Player player) {
-        return playerService.createPlayer(player);
+    public Player createPlayer(@RequestBody RegistrationData registrationData) {
+
+        if(!registrationData.getPassword().equals(registrationData.getPasswordAgain())) {
+            throw new IllegalArgumentException("Die Passwörter stimmen nicht überein.");
+        }
+        registrationData.setPassword(passwordEncoder.encode(registrationData.getPassword()));
+        return playerService.savePlayer(registrationData);
     }
 
     @GetMapping("/{id}")
-    public Player findById(@PathVariable String id) {
+    public Optional<Player> findById(@PathVariable String id) {
```

Comment:
I would write it this way:
```java
public ResponseEntity<Player> findById(@PathVariable String id) {
    return ResponseEntity.of(playerService.findPlayerById(id));
}
```
That way the method returns a status 200 if the `Optional` has a value and otherwise the status 404.